src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -281,6 +281,7 @@ internal bool CommitUnsynchronized()
                 return true;
             }
 
+            Debug.Assert(_writingHead != null);
```

Comment:
Nit: either add a blank line after this or move it down one line to be after the comment